registration-processor/pre-processor/registration-processor-packet-classifier-stage/src/main/java/io/mosip/registration/processor/stages/packetclassifier/tagging/impl/AgeGroupTagGenerator.java

```diff
@@ -0,0 +1,74 @@
+package io.mosip.registration.processor.stages.packetclassifier.tagging.impl;
+
+import io.mosip.kernel.core.exception.BaseCheckedException;
+import io.mosip.registration.processor.core.exception.util.PlatformErrorMessages;
+import io.mosip.registration.processor.packet.storage.utils.Utilities;
+import io.mosip.registration.processor.stages.packetclassifier.dto.FieldDTO;
+import io.mosip.registration.processor.stages.packetclassifier.tagging.TagGenerator;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.stereotype.Component;
+
+@Component
+@ConditionalOnExpression(value = "'${mosip.regproc.packet.classifier.tag-generators}'.contains('MosipAgeGroup')")
+public class AgeGroupTagGenerator implements TagGenerator {
+
+    /** Tag name that will be used while tagging age group */
+    @Value("${mosip.regproc.packet.classifier.tagging.agegroup.tag-name:AGE_GROUP}")
+    private String tagName;
+
+    /** Different age group that should be used for age group tagging */
+    @Value("#{'${mosip.regproc.packet.classifier.tagging.agegroup.group-names:CHILD,ADULT,SENIOR_CITIZEN}'.split(',')}")
```

Comment:
can we not use a value map so the age group as key and range as value?

Reply:
Yes, will change that

Reply:
Changes done